joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/RubySrc2Cpg.scala

```diff
@@ -44,7 +45,12 @@ class RubySrc2Cpg extends X2CpgFrontend[Config] {
 
   private def createCpgAction(cpg: Cpg, config: Config): Unit = {
     File.usingTemporaryDirectory("rubysrc2cpgOut") { tmpDir =>
-      val astGenResult = RubyAstGenRunner(config).execute(tmpDir)
+      val astGenResult = persistedRubyAstGenRunner match {
+        case Some(astGenRunner) =>
+          astGenRunner.execute(tmpDir, config)
+        case None =>
+          Using.resource(RubyAstGenRunner(config))(_.execute(tmpDir))
+      }
```

Comment:
If we create the runner here on demand for the non server scenario, when not always create it here on demand?

And i suggest we let X2CpgFrontend extend `AutoClosable` so that the `close` call can be handled be X2CpgFrontend and not in the individual frontend Main classes.